pkg/apis/core/v1/conversion.go

```diff
@@ -539,3 +546,146 @@ func Convert_core_PersistentVolumeSpec_To_v1_PersistentVolumeSpec(in *core.Persi
 func Convert_v1_PersistentVolumeSpec_To_core_PersistentVolumeSpec(in *v1.PersistentVolumeSpec, out *core.PersistentVolumeSpec, s conversion.Scope) error {
 	return autoConvert_v1_PersistentVolumeSpec_To_core_PersistentVolumeSpec(in, out, s)
 }
+
+// HandleSeccompVersionSkew implements the version skew behavior described in:
+// https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/20190717-seccomp-ga.md#version-skew-strategy
+func HandleSeccompVersionSkew(in *core.Pod, out *v1.Pod) error {
+	// Handle the pod
```

Comment:
we only care about the case where they only provided one of the annotation and the field, right? would this be easier to follow like this:

```go
annotation, hasAnnotation := pod.Annotations[v1.SeccompPodAnnotationKey]

field, hasField := (*core.SeccompProfile)(nil), false
if pod.Spec.SecurityContext != nil && pod.Spec.SecurityContext.SeccompProfile != nil {
  field = pod.Spec.SecurityContext.SeccompProfile
  hasField = true
}

if hasAnnotation != hasField {
  // initialize parents if necessary
  if pod.Annotations == nil {
    pod.Annotations = map[string]string{}
  }
  if pod.Spec.SecurityContext == nil {
    pod.Spec.SecurityContext = &v1.PodSecurityContext{}
  }
  // assign computed values
  pod.Annotations[v1.SeccompPodAnnotationKey], pod.Spec.SecurityContext.SeccompProfile = syncSeccompAnnotationAndField(annotation, field)
}
```

(and then something similar per-container inside VisitContainers)

Reply:
Yes, thank you for the hint. Changed as suggested.